src/test/java/no/regnskap/integration/RegnskapApiIntegration.java

```diff
@@ -0,0 +1,83 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static no.regnskap.TestData.*;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RegnskapApiIntegration {
+    private static String regnskapRootURL;
+
+    @ClassRule
+    public static DockerComposeContainer compose = new DockerComposeContainer<>(new File("src/test/resources/test-compose.yml"))
+        .withExposedService(MONGO_SERVICE_NAME, MONGO_PORT, Wait.forListeningPort())
+        .withExposedService(API_SERVICE_NAME, API_PORT, Wait.forHttp("/ping").forStatusCode(200))
+        .withPull(false);
+
+    @BeforeClass
+    public static void mongoSetup() {
+        CodecRegistry pojoCodecRegistry = fromRegistries(MongoClient.getDefaultCodecRegistry(), fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+        MongoClient mongoClient = new MongoClient(compose.getServiceHost(MONGO_SERVICE_NAME, MONGO_PORT), compose.getServicePort(MONGO_SERVICE_NAME, MONGO_PORT));
+        MongoDatabase mongoDatabase = mongoClient.getDatabase(DATABASE_NAME).withCodecRegistry(pojoCodecRegistry);
+        MongoCollection<RegnskapDB> mongoCollection = mongoDatabase.getCollection(COLLECTION_NAME).withDocumentClass(RegnskapDB.class);
+
+        mongoCollection.insertOne(regnskap2017);
+        mongoCollection.insertOne(regnskap2018);
+
+        regnskapRootURL = "http://" + compose.getServiceHost(API_SERVICE_NAME, API_PORT) + ":" + compose.getServicePort(API_SERVICE_NAME, API_PORT);
+    }
+
+    @Test
+    public void pingTest() throws Exception {
+        String response = simpleGet(regnskapRootURL + "/ping");
+        Assert.assertEquals("RegnskapAPI is available", "pong", response);
+    }
+
+    @Test
+    public void getByOrgnrTest() throws Exception {
+        String response = simpleGet(regnskapRootURL + "/regnskap?orgNummer=orgnummer");
+        Assert.assertEquals(EXPECTED_RESPONSE_ORGNR, response);
+    }
+
+    @Test
+    public void getById() throws Exception {
+        String response2018 = simpleGet(regnskapRootURL + "/regnskaps/" + GENERATED_ID_0.toHexString());
+        String response2017 = simpleGet(regnskapRootURL + "/regnskaps/" + GENERATED_ID_1.toHexString());
+        Assert.assertEquals(buildExpectedDatabaseResponse(GENERATED_ID_0, 2018), response2018);
+        Assert.assertEquals(buildExpectedDatabaseResponse(GENERATED_ID_1, 2017), response2017);
+    }
+
+    private String simpleGet(String address) throws Exception {
+        URL obj = new URL(address);
+        HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+        con.setRequestMethod("GET");
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()));
```

Comment:
try-with-resource?

Reply:
If the test fails it can just as well fail by Exception?

Reply:
try-with-resource would clean up streams, but I do see your point in this being a test   :-)

Reply:
6496aad30a2522388345cd44e160458b9a20e264

Cleaner code and still fails by Exception